logdetective/server.py

```diff
@@ -129,32 +129,36 @@ def submit_text(text: str, max_tokens: int = 0, log_probs: int = 1):
     data = {
             "prompt": text,
             "max_tokens": str(max_tokens),
-            "logprobs": str(log_probs)}
+            "logprobs": str(log_probs),
+            "stream": stream}
 
     try:
         # Expects llama-cpp server to run on LLM_CPP_SERVER_ADDRESS:LLM_CPP_SERVER_PORT
         response = requests.post(
             f"{LLM_CPP_SERVER_ADDRESS}:{LLM_CPP_SERVER_PORT}/v1/completions",
             headers={"Content-Type":"application/json"},
             data=json.dumps(data),
-            timeout=int(LLM_CPP_SERVER_TIMEOUT))
+            timeout=int(LLM_CPP_SERVER_TIMEOUT),
+            stream=stream)
     except requests.RequestException as ex:
         raise HTTPException(
             status_code=400,
             detail=f"Llama-cpp query failed: {ex}") from ex
-
-    if not response.ok:
-        raise HTTPException(
-            status_code=400,
-            detail="Something went wrong while getting a response from the llama server: "
-                f"[{response.status_code}] {response.text}")
-    try:
-        response = json.loads(response.text)
-    except UnicodeDecodeError as ex:
-        LOG.error("Error encountered while parsing llama server response: %s", ex)
-        raise HTTPException(
-            status_code=400,
-            detail=f"Couldn't parse the response.\nError: {ex}\nData: {response.text}") from ex
+    if not stream:
+        if not response.ok:
+            raise HTTPException(
+                status_code=400,
+                detail="Something went wrong while getting a response from the llama server: "
+                    f"[{response.status_code}] {response.text}")
+        try:
+            response = json.loads(response.text)
+        except UnicodeDecodeError as ex:
+            LOG.error("Error encountered while parsing llama server response: %s", ex)
+            raise HTTPException(
+                status_code=400,
+                detail=f"Couldn't parse the response.\nError: {ex}\nData: {response.text}") from ex
+    else:
+        return response
```

Comment:
I'm sure we'll need to tinker with the error handling for the streaming response in future once we'll start using this.